File: iact-check-core/src/main/kotlin/ch/iact/iactcheck/dto/CheckDTO.kt
package ch.iact.iactcheck.dto

data class CheckDTO(
    val id: Long,
    val customerId: Long,
    val title: Map<String, String>,
    val subtitle: Map<String, String>,
    val requiredLanguages: Set<LanguageDTO>,
    val defaultLanguage: LanguageDTO,
    val questionCategories: List<QuestionCategoryDTO>,
    val marketplaceConfig: MarketplaceConfigDTO?,
    val introductionSlideConfiguration: IntroductionSlideConfigurationDTO,
    val emailSubject: Map<String, String>,
    val emailMessage: Map<String, String>,
    val textMessage: Map<String, String>
)

File: iact-check-webapp/src/main/resources/webapp/src/app/shared/dtos/check-dto.ts
import { QuestionCategoryDTO } from './question-category-dto';
import { LanguageDTO } from './language-dto';
import { MarketplaceConfigDTO } from './marketplace-config-dto';
import { IntroductionSlideConfigurationDTO } from './introduction-slide-configuration-dto';

export interface CheckDTO {
  id?: number;
  customerId: number;
  title: { [key: string]: string };
  subtitle: { [key: string]: string };
  requiredLanguages: LanguageDTO[];
  defaultLanguage: LanguageDTO;
  activeFrom: Date;
  activeTo: Date;
  questionCategories: QuestionCategoryDTO[];
  marketplaceConfig?: MarketplaceConfigDTO;
  introductionSlideConfiguration: IntroductionSlideConfigurationDTO;
  emailSubject: { [key: string]: string };
  emailMessage: { [key: string]: string };
  textMessage: { [key: string]: string };
}
